regression/goto-analyzer/pointer-comparison/main.c

```diff
@@ -0,0 +1,15 @@
+#include <assert.h>
+
+int main()
+{
+  int a[2];
+  int b[2];
+
+  int *p = a;
+  int *q = a;
+
+  assert(p - q == 0);
+
+  q = b;
+  assert(p - q == 0);
```

Comment:
I have a feeling that the C standard says this is undefined behaviour.  We probably should support it and so it is reasonable to test for it but until we have a more principled way of handling UB, we should be careful.  If you have to modify the code then a comment noting that this is intentionally UB would be good.